lib/Mojo/mysql/Migrations.pm

```diff
@@ -0,0 +1,231 @@
+  $target //= $self->latest;
+
+  # Unknown version
+  my ($up, $down) = @{$self->{migrations}}{qw(up down)};
+  croak "Version $target has no migration" if $target != 0 && !$up->{$target};
+
+  # Already the right version (make sure migrations table exists)
+  my $db = $self->mysql->db;
+  return $self if $self->_active($db) == $target;
+
+  # Lock migrations table and check version again
+  local @{$db->dbh}{qw(AutoCommit RaiseError)} = (1, 1);
+  my $tx = $db->begin;
+
+  return $self if (my $active = $self->_active($db)) == $target;
+
+  # Up
+  my $sql;
+  if ($active < $target) {
+    $sql = join '',
+      map { $up->{$_} } grep { $_ <= $target && $_ > $active } sort keys %$up;
+  }
+
+  # Down
+  else {
+    $sql = join '',
+      map { $down->{$_} }
+      grep { $_ > $target && $_ <= $active } reverse sort keys %$down;
+  }
+
+  warn "-- Migrate ($active -> $target)\n$sql\n" if DEBUG;
+
+  chop($sql);
+  $db->query($_) for split(';', $sql); 
+  $sql = "update mojo_migrations set version = ? where name = ?;";
+  $db->query($sql, $target, $self->name) and $tx->commit;
+  return $self;
+}
+
+sub _active {
+  my ($self, $db) = @_;
+
+  my $dbh  = $db->dbh;
+  my $name = $self->name;
+  local @$dbh{qw(AutoCommit RaiseError)} = (1, 0);
+  my $results
+    = $db->query('select version from mojo_migrations where name = ?', $name);
+  if (my $next = $results->array) { return $next->[0] }
+
+  local @$dbh{qw(AutoCommit RaiseError)} = (1, 1);
+  $db->query(
+    'create table if not exists mojo_migrations (
+       name    varchar(255) unique,
+       version varchar(255)
+     )'
+  ) if $results->sth->err;
+  $db->query('insert into mojo_migrations values (?, ?)', $name, 0);
+
+  return 0;
+}
+
+1;
+
+=encoding utf8
+
+=head1 NAME
+
+Mojo::mysql::Migrations - Migrations
+
+=head1 SYNOPSIS
+
+  use Mojo::mysql::Migrations;
+
+  my $migrations = Mojo::mysql::Migrations->new(mysql => $mysql);
+  $migrations->from_file('/Users/sri/migrations.sql')->migrate;
+
+=head1 DESCRIPTION
+
+L<Mojo::mysql::Migrations> is used by L<Mojo::mysql> to allow database schemas to
+evolve easily over time. A migration file is just a collection of sql blocks,
+with one or more statements, separated by comments of the form
+C<-- VERSION UP/DOWN>.
+
+  -- 1 up
+  create table messages (message varchar(255));
+  insert into messages values ('I â™¥ Mojolicious!');
+  -- 1 down
+  drop table messages;
+  -- 2 up (...you can comment freely here...)
+  create table stuff (whatever int);
+  -- 2 down
+  drop table stuff;
+
+The idea is to let you migrate from any version, to any version, up and down.
+Migrations are very safe, because they are performed in transactions and only
+one can be performed at a time. If a single statement fails, the whole
+migration will fail and get rolled back. Every set of migrations has a
+L</"name">, which is stored together with the currently active version in an
+automatically created table named C<mojo_migrations>.
+
```

Comment:
I think we should add the warning here. Something like this:

    =head1 CAVEAT
    
    MySQL does not support DDL transactions. Therefore, migrations should be used with extreme caution. Backup your database. You've been warned!